Add deeplink docu for add hotel to trip feature (#2)
FriederikeKr/preview.developer.concur.com: src/api-guides/travel/deeplink-integration-new.md

This deeplink will perform a search for the hotel Travelodge LAX South in El Segundo (CA) and hotels nearby in the 5 mi default radius, with check-in date on November 2nd, 2024 and checkout date on November 3rd, 2024. It uses the Hotel Property ID from HRS as location.

#### <a name="example-one-way"></a> Example: SAP Concur Travel Internal Hotel ID location
```
https://www.concursolutions.com/goto/hotel-shop?checkindate=2024-11-02&checkoutdate=2024-11-03&location=1907825ad1f728ccafb22942d61a2715
```

This deeplink will perform a search for the hotel Econo Lodge Inn & Suites in Arkansas and hotels nearby in the 5 mi default radius, with check-in date on November 2nd, 2024 and checkout date on November 3rd, 2024. It uses a SAP Concur Travel Internal Hotel ID as location.

## Add Hotel to existing Trip

> Users are able to provide a trip ID in a hotel deeplink and omit the check-in and check-out dates. Then the hotel search and subsequent hotel booking will be part of the specified trip, instead of a separate new trip. In contrast to a deeplink without trip ID, the user will end up on the hotel search page with pre-filled search criteria. The check-in and check-out dates are either taken over from the arrival of the first air booking and the departure of the last air booking of the specified trip. In case of a one-way flight the check-out date is taken over from the trip end date. Or the dates fallback to the trip start and end dates of the specified trip. These search criteria can still be edited, before a search can be started.
> Only if a hotel property ID is provided the search is automatically started.

Reviewer: I'd mention the difference between providing a location or not earlier, also IMO simplified a bit the comment about the fallback dates.

```suggestion
Users are able to provide a trip ID in a hotel deeplink and omit the check-in and check-out dates. Then the hotel search and subsequent hotel booking will be part of the specified trip, instead of a separate new trip.

If only a trip ID is provided without a specific location, the user will end up on the hotel search page with pre-filled search criteria based on the trips booking.

If in addition to the trip ID a specific location is provided, the user will directly get to the search results based on this location. The check-in and check-out dates are taken over from the arrival of the first air booking and the departure of the last air booking of the specified trip. In case of a one-way flight the check-out date is taken over from the trip end date. If the trip contains no air booking then the trip start and end dates are used.
```

Author: 👍 I misunderstood that the location type doesn't matter for where the user ends up (pre-filled search from vs. shop results), but the fact whether a location was provided is deciding. makes more sense

#### URI Template

```
https://www.concursolutions.com/goto/hotel-shop?tripid={tripid}&location={location}&locationsource={locationsource}&distance={distance}&unit={unit}
```

#### Example of Usage
```
https://www.concursolutions.com/goto/hotel-shop?tripid=9be19dda-bf33-4110-ac48-bf840878cd3c&location=50.050686,8.565997&distance=15&unit=mi
```

#### Query Parameters

|Name            |Type     |Format                                                                                                                                             |Description|
|----------------|---------|---------------------------------------------------------------------------------------------------------------------------------------------------|-----------|
|`tripid`        |`string` |`UUID`                                                                                                                                             |**Required** <br>Unique identifier of the trip, formatted as a UUID.|

> |`location`      |`string` |[Location Format](/api-guides/travel/deeplink-integration.html#location-format-hotel)                                                              |**Required** <br>Location of the search center with default search radius of 5 mi.|

Reviewer: ```suggestion
|`location`      |`string` |[Location Format](/api-guides/travel/deeplink-integration.html#location-format-hotel)                                                              |(Optional) <br>Location of the search center with default search radius of 5 mi.|
```

> |`locationsource`|`string` |one of {`northstar`, `giata`, `leonardo`, `amadeus`, `sabre`, `galileo`, `cwt`, `expedia`, `hrs`, `booking.com`, `omnibees`}                       |**Required if location is a Hotel Property ID**. <br>Source of the Hotel Property ID. For geo-coordinates and SAP Concur Travel Internal Hotel ID, omit this query parameter.|

Reviewer: ```suggestion
|`locationsource`|`string` |one of {`northstar`, `giata`, `leonardo`, `amadeus`, `sabre`, `galileo`, `cwt`, `expedia`, `hrs`, `booking.com`, `omnibees`}                       |**Required if location provided and is a Hotel Property ID**. <br>Source of the Hotel Property ID. For geo-coordinates and SAP Concur Travel Internal Hotel ID, omit this query parameter.|
```

|`distance`      |`integer`|one of {`1`, `2`, `3`, `4`, `5`, `10`, `15`, `20`, `25`, `30`, `35`, `40`, `45`, `50`, `55`, `60`, `65`, `70`, `75`, `80`, `85`, `90`, `95`, `100`}|(Optional) <br> Distance around the `location`. <br>If not set, it will default to 5 mi `distance` and `radius` as search radius.|
|`unit`          |`string` |`km` or `mi`                                                                                                                                       |(Optional) <br> Unit for the `distance`. <br>If not set, the value will be taken from users profile - under Profile Settings, System Settings, **mile/km**.|

> #### <a name="example-one-way"></a> Example: Hotel Property ID location

Reviewer: Name is wrong

```
https://www.concursolutions.com/goto/hotel-shop?tripid=a7f172cd-8587-48d5-8533-5ddb7ff5c26c&location=708752&locationsource=hrs
```

> This deeplink will perform a search for the hotel Travelodge LAX South in El Segundo (CA) and hotels nearby in the 5 mi default radius. It uses the Hotel Property ID from HRS as location. The check-in and check-out dates are taken from the corresponding trip with trip id `a7f172cd-8587-48d5-8533-5ddb7ff5c26c`. Either the trip contains air bookings, then the arrival date of the first flight will be used as check-in date for the hotel search and the departure date of the last flight will be used as check-out date, or the trip start and end dates will be used as check-in and check-out dates, respectively.

Reviewer: To highlight the difference between providing a location or not
```suggestion
This deeplink will perform a search for the hotel Travelodge LAX South in El Segundo (CA) and hotels nearby in the 5 mi default radius. It uses the Hotel Property ID from HRS as location. The check-in and check-out dates are taken from the corresponding trip with trip id `a7f172cd-8587-48d5-8533-5ddb7ff5c26c`. Either the trip contains air bookings, then the arrival date of the first flight will be used as check-in date for the hotel search and the departure date of the last flight will be used as check-out date, or the trip start and end dates will be used as check-in and check-out dates, respectively. The user will get directly to the search results.
```

> #### <a name="example-one-way"></a> Example: Geo-coordinate location

Reviewer: Also here name needs adjustment

```

> https://www.concursolutions.com/goto/hotel-shop?tripid=a7f172cd-8587-48d5-8533-5ddb7ff5c26c&location=48.85694273527786,2.3501079080340315

Reviewer: I'd either in addition or instead of the coordinate example provide an example with just a trip ID and no location (maybe order of these examples could match the order we explain it above).

```

> This deeplink will pre-fill the hotel search form with Paris, in particular the coordinates `48.85694273527786,2.3501079080340315` (**todo: test how geo-coordinates are displayed in hotel search form**), as the search center with a 5 mi radius around it and with check-in and check-out dates taken from the trip corresponding to the trip id `a7f172cd-8587-48d5-8533-5ddb7ff5c26c`. Either the arrival date of the first flight will be used as check-in date and the departure of the last flight will be used as check-out date, or the trip start and end dates will be used, respectively.

Reviewer: This will also directly start a search as a location is provided, therefore the coordinate display is the same as with the regular hotel deeplink (so not sure if it requires a specific mention).

|`locationsource`|`string` |one of {`northstar`, `giata`, `leonardo`, `amadeus`, `sabre`, `galileo`, `cwt`, `expedia`, `hrs`, `booking.com`, `omnibees`}                       |**Required if location is provided and is a Hotel Property ID**. <br>Source of the Hotel Property ID. For geo-coordinates and SAP Concur Travel Internal Hotel ID, omit this query parameter.|
|`distance`      |`integer`|one of {`1`, `2`, `3`, `4`, `5`, `10`, `15`, `20`, `25`, `30`, `35`, `40`, `45`, `50`, `55`, `60`, `65`, `70`, `75`, `80`, `85`, `90`, `95`, `100`}|(Optional) <br> Distance around the `location`. <br>If not set, it will default to 5 mi `distance` and `radius` as search radius.|
|`unit`          |`string` |`km` or `mi`                                                                                                                                       |(Optional) <br> Unit for the `distance`. <br>If not set, the value will be taken from users profile - under Profile Settings, System Settings, **mile/km**.|

#### <a name="example-add-to-trip"></a> Example: Only trip ID
```
https://www.concursolutions.com/goto/hotel-shop?tripid=a7f172cd-8587-48d5-8533-5ddb7ff5c26c
```

> This deeplink will pre-fill the hotel search form to add a booking to the trip with id `a7f172cd-8587-48d5-8533-5ddb7ff5c26c`. The location, check-in and check-out dates will be derived from the existing bookings of the trip. By default the first booking of the trip will be used, but the user can select another one in the form itself.

Reviewer: That will avoid us having to adapt the docu in case something with the hotel search proposals will change.
```suggestion
This deeplink will redirect the user to the hotel search form to add a booking to the trip with id `a7f172cd-8587-48d5-8533-5ddb7ff5c26c`. Search proposals based on the content of the trip are available in the search form.
```
